onegov/core/tests/test_utils.py

```diff
@@ -137,6 +137,13 @@ def test_linkify():
         == '<a href="mailto:info@example.org">info@example.org</a><br>'
 
 
+def test_linkify_with_phone():
```

Comment:
I found an instance where this function fails (note the additional newline):

```python
def test_linkify_with_phone():
    tel = 'Tel. +41 41 728 33 11'
    result = 'Tel. <a href="tel:+41 41 728 33 11">+41 41 728 33 11</a>'
    assert utils.linkify(tel, escape=False) == result
    assert utils.linkify(tel, escape=True) == result

    tel = 'Tel. +41 41 728 33 11\n'
    result = 'Tel. <a href="tel:+41 41 728 33 11">+41 41 728 33 11</a>'
    assert utils.linkify(tel, escape=False) == result
    assert utils.linkify(tel, escape=True) == result
```

Looks like the `is_valid_length` function is fed with a string that includes newlines.

Reply:
The phone_ch regex pattern includes also newlines which it shouldn't. I excluded newline from matching the phone, when matched replace all \s and then check the length. Assert in the handle_match that the match is not ending with a newline.